Libraries/LibWeb/Clipboard/ClipboardItem.cpp

```diff
@@ -0,0 +1,209 @@
+
+    // 3. Set this's clipboard item to a new clipboard item.
+    auto clipboard_item = realm.create<ClipboardItem>(realm);
+
+    // 4. Set this's clipboard item's presentation style to options["presentationStyle"].
+    clipboard_item->m_presentation_style = options.presentation_style;
+
+    // 5. Let types be a list of DOMString.
+    Vector<String> types;
+
+    // 6. For each (key, value) in items:
+    for (auto const& [key, value] : items) {
+        // 2. Let isCustom be false.
+        bool is_custom = false;
+
+        // 3. If key starts with `"web "` prefix, then:
+        auto key_without_prefix = key;
+        if (key.starts_with_bytes(WEB_CUSTOM_FORMAT_PREFIX)) {
+            // 1. Remove `"web "` prefix and assign the remaining string to key.
+            key_without_prefix = MUST(key.substring_from_byte_offset(WEB_CUSTOM_FORMAT_PREFIX.length()));
+
+            // 2. Set isCustom to true.
+            is_custom = true;
+        }
+
+        // 5. Let mimeType be the result of parsing a MIME type given key.
+        auto mime_type = MimeSniff::MimeType::parse(key_without_prefix);
+
+        // 6. If mimeType is failure, then throw a TypeError.
+        if (!mime_type.has_value()) {
+            return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, MUST(String::formatted("Invalid MIME type: {}", key)) };
+        }
+
+        String mime_type_serialized = mime_type->serialized();
+
+        // 7. If this's clipboard item's list of representations contains a representation whose MIME type
+        //    is mimeType and whose [representation/isCustom] is isCustom, then throw a TypeError.
+        auto existing = clipboard_item->m_representations.find_if([&](auto const& item) {
+            return item.mime_type == mime_type_serialized && item.is_custom == is_custom;
+        });
+        if (!existing.is_end()) {
+            return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, MUST(String::formatted("Duplicate MIME type: {}", key)) };
+        }
+
+        // 11. Let mimeTypeString be the result of serializing a MIME type with mimeType.
+        // 12. If isCustom is true, prefix mimeTypeString with `"web "`.
+        String mime_type_string = is_custom ? MUST(String::formatted("{}{}", WEB_CUSTOM_FORMAT_PREFIX, mime_type_serialized)) : mime_type_serialized;
+
+        // 13. Add mimeTypeString to types.
+        types.append(mime_type_string);
+
+        // 1. Let representation be a new representation.
+        // 4. Set representation’s isCustom flag to isCustom.
+        // 8. Set representation’s MIME type to mimeType.
+        // 9. Set representation’s data to value.
+        // 10. Append representation to this's clipboard item's list of representations.
+        Representation representation { move(mime_type_serialized), is_custom, value };
+        clipboard_item->m_representations.append(representation);
```

Comment:
`empend` can be used here instead of creating the object then appending it - like this: `clipboard_item->m_representations.empend(move(mime_type_serialized), is_custom, value)`